projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -543,6 +551,49 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
         return DataUtil.sort(cloneArray(data), expressions, this.grid.sortStrategy, this.grid);
     }
 
+    public pin_row(rowID: any, index?: number, row?: RowType): void {
+        const grid = (this.grid as any);
+        if (grid._pinnedRecordIDs.indexOf(rowID) !== -1) {
+            return;
+        }
+        const eventArgs = this.get_pin_row_event_args(rowID, index, row, true);
+        grid.rowPinning.emit(eventArgs);
+
+        if (eventArgs.cancel) {
+            return;
+        }
+        const insertIndex = typeof eventArgs.insertAtIndex === 'number' ? eventArgs.insertAtIndex : grid._pinnedRecordIDs.length;
+        grid._pinnedRecordIDs.splice(insertIndex, 0, rowID);
+    }
+
+    public unpin_row(rowID: any): void {
+        const grid = (this.grid as any);
+        const index = grid._pinnedRecordIDs.indexOf(rowID);
+        if (index === -1) {
+            return;
+        }
+        const eventArgs = this.get_pin_row_event_args(rowID, null , null, false);
```

Comment:
Why two null values are passed for the `index` and `row` param? Both are available.